### PR TITLE
Add --keep-file-path

### LINKED ISSUE #611: Option to preserve directory paths
For Android, some users have requested to preserve the original directory paths. Currently Flank downloads files to a flattened directory which requires unique file names.

I'm looking for feedback on this as it seems like an edge case that's not super useful. Files should be uniquely named, especially images.

### PR DESCRIPTION
**Pull request by tahirhajizada**
Fix #611

### CODE REVIEW

**Comment by bootstraponline** [comment]
Thanks for submitting a pull request! I'll take a look soon.

File: README.md [hunk 1]
```diff
@@ -311,6 +311,10 @@ flank:
 
   ## Local folder to store the test result. Folder is DELETED before each run to ensure only artifacts from the new run are saved.
   # local-result-dir: flank
+
+  ## Downloaded files preserves the original path of file. Required when file names are not unique.
+  ## Default: false  
+  # preserve-original-path: false
```
Comment (bootstraponline) [inline README.md]: Thanks for working on this! What do you think of `keep-file-path` for the key name?

File: test_runner/src/main/kotlin/ftl/util/ObjPath.kt [hunk 2]
```diff
@@ -16,12 +17,16 @@ data class ObjPath(
             val objName = parsed.getName(0).toString()
             val shardName = parsed.getName(1).toString()
             val deviceName = parsed.getName(2).toString()
+            val filePathName = if (parsed.nameCount > 4) {
+                parsed.subpath(3, parsed.nameCount - 1).toString()
```
Comment (bootstraponline) [inline test_runner/src/main/kotlin/ftl/util/ObjPath.kt]: What do you think of `parsed.parent.drop(3).joinToString("/")` as a way to simplify the `parsed.subpath` logic?

**Comment by bootstraponline** [comment]
I'm happy to make the suggested changes if that'd be easier.

**Comment by tahirhajizada** [comment]
I am ok with your suggestions. Makes sense. It would be nice if you apply them or I can do after 9-10 hours :)

**Comment by bootstraponline** [comment]
Sounds good. I'll apply the fixes in ~4 hours after work.

**Comment by bootstraponline** [comment]
I updated, tested, and merged your pull request! Nice work.

I released Flank v8.1.0 with the feature and updated the change log.

Thanks for contributing.

**Comment by tahirhajizada** [comment]
Thank you for the changes and release. Welcome.

**Pull request merged**